### PR TITLE
feat: Add delegation portal

### PR DESCRIPTION
**Pull request by samuveth**
### Issues

Fixes https://github.com/snapshot-labs/snapshot/issues/3893
Fixes https://github.com/snapshot-labs/snapshot/issues/3892

### Changes 

1. Add delegation section to settings
2. Add new space page for delegation portal
3. Enable compound governor delegation standard 

### How to test

1. https://snapshot-m64zxm8pf-snapshot.vercel.app/#/testsnap.eth

OR

1. Go to your space settings -> Delegation
5. Set contract to `0x1f9840a85d5af5bf1d1762f925bdaddc4201f984` and URL to `https://api.thegraph.com/subgraphs/name/arr00/uniswap-governance-v2`
6. Save settings
7. You should see a new 

### Self-review checklist
- [ ] I have performed a full self-review of my changes
- [ ] I have tested my changes on a preview deployment
- [ ] I have tested my changes on a custom domain
- [ ] I have run end-to-end tests `yarn cypress:test:e2e`, and they have passed

### Additional notes or considerations

Spaces that have a delegation strategy and delegation setting will see both `Delegate` and `Delegates` in the space sidebar. Snapshot "Delegate" is working exactly the same as before on the UI

### CODE REVIEW

**Comment by j0hnfl0w** [comment]
whats happened with this button @samuveth ?
image

**Comment by j0hnfl0w** [comment]
How do we get to spaceDelegates route from space route?

File: src/components/SpaceDelegatesListItem.vue [hunk 1]
```diff
@@ -0,0 +1,61 @@
+<script setup lang="ts">
+import { Delegate, Profile, ExtendedSpace } from '@/helpers/interfaces';
+
+const { domain } = useApp();
```
Comment (j0hnfl0w) [inline src/components/SpaceDelegatesListItem.vue:4]: Maybe composables should go after props?

**Comment by j0hnfl0w** [comment]
> How do we get to spaceDelegates route from space route?

ah i see there is no hasDelegationStrategy, but direct access to route works anyway?

File: src/views/SpaceDelegates.vue [hunk 2]
```diff
@@ -0,0 +1,53 @@
+<script setup lang="ts">
+import { ExtendedSpace } from '@/helpers/interfaces';
+import { useInfiniteScroll } from '@vueuse/core';
+
+const {
```
Comment (j0hnfl0w) [inline src/views/SpaceDelegates.vue:5]: Same here composables should go after props.

I know its a draft, just some comments.

**Comment by samuveth** [comment]
This PR is not even close to ready for a review :D @j0hnfl0w  Will request when ready

**Comment by ChaituVR** [comment]
@samuveth Can resolve conflicts? want to try the preview to save delegation settings so I can test my changes on the hub

**Comment by samuveth** [comment]
@ChaituVR done

File: src/components/SpaceDelegatesListItem.vue [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      </div>
+    </button>
+
+    <div class="mt-2 h-full">
+      <template v-if="delegate?.statement">
+        <span class="line-clamp-3">
+          {{ delegate.statement }}
+        </span>
+        <span class="cursor-pointer text-skin-link" @click="emit('clickUser')">
+          Lean more
```
Comment (ChaituVR) [inline src/components/SpaceDelegatesListItem.vue:45]:
```suggestion
          Learn more
```

**Review by ChaituVR — CHANGES_REQUESTED** [review]
Looks like `delegate_in` not working? 

when i search with my name, 
https://snapshot-git-feat-delegate-snapshot.vercel.app/#/testsnap.eth/delegates?search=chaitu.eth

it get all statements, instead of one 
image

**Comment by ChaituVR** [comment]
Once we click "save" in the "Edit statement"  modal, is it possible to reload the delegate statement?

**Comment by ChaituVR** [comment]
"Edit statement" modal keeps the state,
If we modify something and close the modal without saving, it keeps the edited values. 

https://github.com/snapshot-labs/snapshot/assets/15967809/3780249a-113c-4159-a688-e1330a7c75f5

**Comment by samuveth** [comment]
> Once we click "save" in the "Edit statement" modal, is it possible to reload the delegate statement?

@ChaituVR If I understand you correctly, this is already the case.

> Looks like delegate_in not working?

Yes, is this a issue on the UI?

**Comment by ChaituVR** [comment]
> Yes, is this a issue on the UI?

Yep looks like so, it gets all statements instead of one on-screen if we go here,
https://bafybeih3lr32fpa2ye5lzdm6hphsbajhpzcbxz6aevogkczpynfwqs5osq.on.fleek.co/#/testsnap.eth/delegates?search=chaitu.eth and check network tab for statements query, it fetch all statements, (there is no `delegete_in`

> If I understand you correctly, this is already the case.

Nope. It just shows a "statement updated" message. we have to reload to see the new statement, i mean the delegate card is not updating with the new one
image

**Comment by ChaituVR** [comment]
Also i still see this https://github.com/snapshot-labs/snapshot/pull/3894#issuecomment-1600927694

**Comment by samuveth** [comment]
> Also i still see this [#3894 (comment)](https://github.com/snapshot-labs/snapshot/pull/3894#issuecomment-1600927694)

Not sure why we would want to reset it. Imagine the user accidentally closes the modal, then he would lose all his edits

**Comment by samuveth** [comment]
Thanks for clarifying @ChaituVR, should be fixed now.

**Comment by ChaituVR** [comment]
> Not sure why we would want to reset it. Imagine the user accidentally closes the modal, then he would lose all his edits

Agree 😄 then its a feature, not a bug
Maybe we should ask before closing the modal, but not required for now

**Review by aurelianoB — APPROVED** [review]
tACK

**Comment by ChaituVR** [comment]
The scroll is too weird here
image

Instead I think we should have scroll/overflow this section only 
Untitled

**Comment by ChaituVR** [comment]
These scroll bar looks bad too 
image
image

**Comment by ChaituVR** [comment]
In the edit statement, I think we should have a cancel button 
image
to go back to this view 
image

Now, we only have a close button, which closes the modal entirely

**Comment by ChaituVR** [comment]
image

I don't think we will ever have the option to delegate globally, specially for comp like gov, better to make it a description instead of a select menu

**Comment by ChaituVR** [comment]
Like I mentioned here https://github.com/snapshot-labs/snapshot/pull/3894#pullrequestreview-1490857665

If you go to https://snapshot-git-feat-delegate-snapshot.vercel.app/#/testsnap.eth/delegates?search=chaitu.eth
and open the network tab, look for the `graphql` request we are sending, the statement doesn't use `delegate_in` at all, every time it tries to get all delegations of the space, this will not scale/brake many things when we have a lot of statements

Another example is, go here https://snapshot-git-feat-delegate-snapshot.vercel.app/#/testsnap.eth/delegates
look for requests that are being sent out, every time we hit the pagination, it will get all the statements

**Comment by ChaituVR** [comment]
While the button is disabled, can we display different colors like on the proposal creation page

https://github.com/snapshot-labs/snapshot/assets/15967809/7b3bf098-e56f-4b4c-bc49-b1aff1759100

In the proposal creation page
https://github.com/snapshot-labs/snapshot/assets/15967809/ce84861a-c89d-449f-b9af-764fcfcda252

File: src/components/SpaceDelegatesProfileModal.vue [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+                class="text-xs"
+                @click="
+                  emit('close');
+                  showModalStatement = true;
+                "
+              />
+            </BaseButtonIcon>
+          </div>
+          <p class="text-skin-heading">
+            <BaseMarkdown :body="getStatementStatement(address)!" />
```
Comment (ChaituVR) [inline src/components/SpaceDelegatesProfileModal.vue:112]: This is new, what will happen if we add `!` after the function 🤔 is it vue specific?

File: src/style.scss [hunk 5]
```diff
@@ -118,7 +118,7 @@
 }
 
 .tune-button {
-  @apply rounded-full;
+  @apply h-[42px] rounded-full;
```
Comment (ChaituVR) [inline src/style.scss:121]: Any reason for this? It is 42px without this too, 

https://github.com/snapshot-labs/tune/blob/db2f6f25fa29c7af7da25b0a3ba02c8fe8486c9b/src/style.scss#L69

File: src/components/SpaceDelegatesProfileModal.vue (hunk 4 above)
Comment (samuveth) [reply src/components/SpaceDelegatesProfileModal.vue:112]: https://stackoverflow.com/questions/42273853/in-typescript-what-is-the-exclamation-mark-bang-operator-when-dereferenci

Comment (samuveth) [reply src/components/SpaceDelegatesProfileModal.vue:112]: There is a check above `<div v-if="getStatementStatement(address)">` but typescript doesn't pick that up for some reason. The `!` is just telling typescript that it is always `non-null`

**Comment by samuveth** [comment]
> Like I mentioned here [#3894 (review)](https://github.com/snapshot-labs/snapshot/pull/3894#pullrequestreview-1490857665)
> 
> If you go to https://snapshot-git-feat-delegate-snapshot.vercel.app/#/testsnap.eth/delegates?search=chaitu.eth and open the network tab, look for the `graphql` request we are sending, the statement doesn't use `delegate_in` at all, every time it tries to get all delegations of the space, this will not scale/brake many things when we have a lot of statements
> 
> Another example is, go here https://snapshot-git-feat-delegate-snapshot.vercel.app/#/testsnap.eth/delegates look for requests that are being sent out, every time we hit the pagination, it will get all the statements

I can not reproduce this anymore, I think you are using an old deployment for testing.

Latest: https://bafybeidkfa4mi6tku75ul6krdpjclgklqisbodurdc2uhxdafdntxc3txm.on.fleek.co/#/testsnap.eth/delegates

image

**Comment by samuveth** [comment]
> While the button is disabled, can we display different colors like on the proposal creation page
> 
>  Jun-30-2023.13-22-52.mp4 
> In the proposal creation page https://github.com/snapshot-labs/snapshot/assets/15967809/ce84861a-c89d-449f-b9af-764fcfcda252

The point is to have it disabled so that when the user clicks it he will see the validation errors

File: src/components/SpaceDelegatesProfileModal.vue (hunk 4 above)
Comment (ChaituVR) [reply src/components/SpaceDelegatesProfileModal.vue:112]: Oh i see, i get this error from `eslint-plugin-vue` so i asked

```
[vue/no-parsing-error]
Parsing error: Unexpected token !.eslint-plugin-vue
```

**Review by ChaituVR — APPROVED** [review]
tAck

**Pull request merged**